deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/WritePathAttributeHandler.java

```diff
@@ -0,0 +1,83 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.server.deployment.scanner;
+
+import java.io.File;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.controller.services.path.PathManager;
+import org.jboss.as.controller.services.path.PathManagerService;
+import org.jboss.as.server.deployment.scanner.logging.DeploymentScannerLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+
+/**
+ * @author wangc
+ *
+ */
+public class WritePathAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+    static final WritePathAttributeHandler INSTANCE = new WritePathAttributeHandler();
+
+    public WritePathAttributeHandler() {
+        super(DeploymentScannerDefinition.PATH);
+    }
+
+    @Override
+    protected void validateUpdatedModel(final OperationContext context, final Resource resource)
+            throws OperationFailedException {
+
+        final ModelNode model = resource.getModel();
+        final ModelNode path = model.get(CommonAttributes.PATH);
+
+        final ServiceController<PathManager> controller = (ServiceController<PathManager>) context.getServiceRegistry(false)
+                .getService(PathManagerService.SERVICE_NAME);
+        if (controller != null) {
+            File relativePath;
+            if (model.hasDefined(CommonAttributes.RELATIVE_TO)) {
+                final ModelNode relativeToPath = model.get(CommonAttributes.RELATIVE_TO);
+                final String pathName = controller.getValue().resolveRelativePathEntry(path.asString(),
+                        relativeToPath.asString());
+                relativePath = new File(pathName);
+            } else {
+                relativePath = new File(path.asString());
+            }
+
+            if (!relativePath.exists()) {
+                throw new OperationFailedException(
+                        DeploymentScannerLogger.ROOT_LOGGER.directoryIsNonexistent(relativePath.getAbsolutePath()));
+            }
+            if (!relativePath.isDirectory()) {
+                throw new OperationFailedException(
+                        DeploymentScannerLogger.ROOT_LOGGER.isNotADirectory(relativePath.getAbsolutePath()));
+            }
+            if (!relativePath.canWrite()) {
+                throw new OperationFailedException(
+                        DeploymentScannerLogger.ROOT_LOGGER.directoryIsNotWritable(relativePath.getAbsolutePath()));
+            }
+        }
+    }
```

Comment:
This would need to be done via adding a step. This ensures that if this write-attribute is part of a composite operation (CLI batch or console update) then the validation does not occur until the other steps in the composite have had a chance to execute, e.g. a step that changes the relative-to attribute as well.